src/Microsoft.AspNet.OData.Shared/DefaultEdmPatchMethodHandler.cs

```diff
@@ -0,0 +1,170 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="DefaultEdmODataAPIHandler.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// This is the default patch handler for non-CLR types. This class has default get, create and update
+    /// methods that are used to patch an original collection when the collection is provided.
+    /// </summary>
+    internal class DefaultEdmODataAPIHandler : EdmODataAPIHandler
+    {
+        private IEdmEntityType entityType;
+        private ICollection<IEdmStructuredObject> originalList;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DefaultEdmODataAPIHandler"/> class.
+        /// </summary>
+        /// <param name="originalList">Original collection of the type which needs to be updated.</param>
+        /// <param name="entityType">The Edm entity type of the collection.</param>
+        public DefaultEdmODataAPIHandler(ICollection<IEdmStructuredObject> originalList, IEdmEntityType entityType)
+        {
+            Debug.Assert(entityType != null, "entityType != null");
+
+            this.entityType = entityType;
+            this.originalList = originalList ?? new List<IEdmStructuredObject>();
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override ODataAPIResponseStatus TryGet(IDictionary<string, object> keyValues, out IEdmStructuredObject originalObject, out string errorMessage)
+        {
+            ODataAPIResponseStatus status = ODataAPIResponseStatus.Success;
+            errorMessage = string.Empty;
+            originalObject = null;
+
+            Debug.Assert(keyValues != null, "keyValues != null");
+
+            try
```

Comment:
Looking at the implementation of `GetFilteredItem` method, it doesn't seem to  like there's any statement there that would cause an exception to be thrown. Why do we need a `try...catch` here?

Reply:
We don't have control of how the `originalList` and `keyvalues` look like. The `try...catch` ensures we catch any exception that may arise.